include/nil/marshalling/algorithms/unpack.hpp

```diff
@@ -0,0 +1,155 @@
+            status = val.write(buffer_begin, result.size());
+
+            return result;
+        }
+
+        /*!
+         * @brief
+         *
+         * @ingroup marshalling_algorithms
+         *
+         * @tparam TEndian
+         * @tparam OutputWordType A compatible with std::is_integral type
+         * @tparam TInput std::is_arithmetic type, not a marshalling type. For example, `int`, `uint8_t` or `double`
+         *
+         * @param val
+         * @param status
+         *
+         * @return
+         */
+        template<typename TEndian, typename OutputWordType, typename TInput>
+        typename std::enable_if<is_compatible<TInput>::value
+                                    && std::is_arithmetic<TInput>::value
+                                    && std::is_integral<OutputWordType>::value,
+                                std::vector<OutputWordType>>::type
+            unpack(TInput val, status_type &status) {
+
+            using marshalling_type = typename is_compatible<TInput>::template type<TEndian>;
+
+            marshalling_type m_val = marshalling_type(val);
+            std::vector<OutputWordType> result(m_val.length());
+            typename std::vector<OutputWordType>::iterator buffer_begin = result.begin();
+            status = m_val.write(buffer_begin, result.size());
+
+            return result;
+        }
+
+        /*!
+         * @brief
+         *
+         * @ingroup marshalling_algorithms
+         *
+         * @tparam TEndian
+         * @tparam OutputWordType A compatible with std::is_integral type
+         * @tparam TContainer A compatible with nil::detail::is_container container type.
+         *
+         * @param val
+         * @param status
+         *
+         * @return
+         */
+        template<typename TEndian, typename OutputWordType, typename TContainer>
+        typename std::enable_if<is_compatible<TContainer>::value
+                                    // && nil::detail::is_container<TContainer>::value
+                                    && (!std::is_arithmetic<TContainer>::value)
+                                    && std::is_integral<OutputWordType>::value,
+                                std::vector<OutputWordType>>::type
+            unpack(TContainer val, status_type &status) {
+
+            // static_assert(std::is_arithmetic<typename TContainer::value_type>::value);
+
+            using marshalling_type = typename is_compatible<TContainer>::template type<TEndian>;
+            using marshalling_internal_type = typename marshalling_type::element_type;
+
+            typename std::conditional<is_compatible<TContainer>::fixed_size, 
+                nil::marshalling::container::static_vector<marshalling_internal_type, marshalling_type::max_length()>, 
+                std::vector<marshalling_internal_type>>::type values;
+            for (const auto &val_i : val) {
+                values.push_back(marshalling_internal_type(val_i));
```

Comment:
Can be replaced with `values.emplace_back(val_i);`

Reply:
Thanks! Replaced.